### PR TITLE
Restructure cmake lists

### PR DESCRIPTION
**Pull request by CCInc**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,29 +4,30 @@ copc-lib is a library which provides an easy-to-use reader and writer interface
...
+sudo cmake --install .
 ```
 
 ### Python
 
 ```bash
-git clone --recurse-submodules https://github.com/RockRobotic/copc-lib.git
+git clone https://github.com/RockRobotic/copc-lib.git
 pip install ./copc-lib
 ```
```
Comment (leo-stan) [inline README.md:33]: Should we provide instruction on how to run the example files in a different project?

File: setup.py [hunk 2]
```diff
@@ -47,18 +47,14 @@ def build_extension(self, ext):
...
             cmake_args += [item for item in os.environ["CMAKE_ARGS"].split(" ") if item]
 
-        # In this example, we pass in the version to C++. You might not need to.
-        cmake_args += [
-            "-DEXAMPLE_VERSION_INFO={}".format(self.distribution.get_version())
-        ]
-
         if self.compiler.compiler_type != "msvc":
             # Using Ninja-build since it a) is available as a wheel and b)
             # multithreads automatically. MSVC would require all variables be
```
Comment (leo-stan) [inline setup.py:60]: I wonder if we can pull the version from CMakelists to avoid discrepancies?

**Review by leo-stan — APPROVED** [review]
My mind is blown indeed, you shall get the title of Cmake Guru :P Bloody good job mate!

File: README.md [hunk 3]
```diff
@@ -44,15 +45,15 @@ find_package(copc-lib REQUIRED)
 find_package(lazperf REQUIRED)
```
[the remark this answers is not in the document]
Comment (CCInc) [reply README.md:45]: maybe?

File: setup.py (hunk 2 above)
Comment (CCInc) [reply setup.py:60]: they both get updated by ReleaseIt. we could in theory parse the cmakelist to pull the version, or make a seperate version file. but I don't care to do that

**Pull request merged**